src/main/java/duke/core/Duke.java

```diff
@@ -0,0 +1,339 @@
+package duke.core;
+
+import java.util.ArrayList;
+import java.util.Scanner;
+
+import duke.databse.Database;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.Todo;
+
+
+public class Duke {
+
+    private Database database;
+    private ArrayList<Task> taskList;
+    private UI ui = new UI();
+
+    /**
+     * Constructor
+     * @param filePath
+     */
+    public Duke(String filePath) {
+        database = new Database(filePath);
+    }
+
+
+    /**
+     * search for a task
+     * @param tasklist
+     * @param keyword
+     * @return
+     */
+    public ArrayList<Task> searchTask(ArrayList<Task> tasklist, String keyword) {
+        ArrayList<Task> result = new ArrayList<>();
+        for (Task task: tasklist) {
+            if (task.searchKeyword(keyword)) {
+                result.add(task);
+            }
+        }
+        return result;
+    }
+
+    /**
+     * start the program
+     */
+    public void run() {
+
+        ArrayList<Task> task = database.getData();
+        int taskNum = task.size();
+        String indentation = "       ";
+        final String LIST = "list";
+        final String BYE = "bye";
+        final String DONE = "done";
+        final String DELETE = "delete";
+        final String FIND = "find";
+
+
+        boolean isEnd = false;
+        System.out.println(ui.logo);
+        System.out.println(ui.line);
+        System.out.println(ui.greeting);
+        System.out.println(ui.line);
+        Scanner scanner = new Scanner(System.in);
+
+
+        while (!isEnd) {
+            String keywords = scanner.nextLine();
+            String[] keyword = keywords.split(" ");
+
+
+            switch(keyword[0]) {
+            case FIND:
```

Comment:
I like how you layout your switch structure nicely and clearly. Readability enhanced!